matchzoo/models/bimpm.py

```diff
@@ -0,0 +1,420 @@
+"""An implementation of BiMPM Model."""
+import typing
+
+import torch
+import torch.nn as nn
+from torch.nn import functional as F
+
+from matchzoo.engine.param_table import ParamTable
+from matchzoo.engine.param import Param
+from matchzoo.engine.base_model import BaseModel
+from matchzoo.dataloader import callbacks
+
+
+class BiMPM(BaseModel):
+    """
+    BiMPM Model.
+
+    Reference:
+    - https://github.com/galsang/BIMPM-pytorch/blob/master/model/BIMPM.py
+
+    Examples:
+        >>> model = BiMPM()
+        >>> model.guess_and_fill_missing_params(verbose=0)
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params(
+            with_embedding=True,
+            with_multi_layer_perceptron=False
+        )
+        params.add(Param(name='mask_value', value=0,
+                         desc="The value to be masked from inputs."))
+        params.add(Param(name='dropout', value=0.2,
+                         desc="Dropout rate."))
+        params.add(Param(name='hidden_size', value=100,
```

Comment:
For hyper-parameters which can be tuned in the auto ML component, you can add `hyper_space ` in the Param definition. You can refer to 'matchzoo/models/conv_knrm.py' to see how to add them.